crates/router/src/core/payment_methods/cards.rs

```diff
@@ -217,24 +221,143 @@ pub async fn create_payment_method(
-    _payment_method_data: Option<Encryption>,
-    _key_store: &domain::MerchantKeyStore,
-    _connector_mandate_details: Option<serde_json::Value>,
-    _status: Option<enums::PaymentMethodStatus>,
-    _network_transaction_id: Option<String>,
-    _storage_scheme: MerchantStorageScheme,
-    _payment_method_billing_address: Option<Encryption>,
-    _card_scheme: Option<String>,
+    state: &routes::SessionState,
+    req: &api::PaymentMethodCreate,
+    customer_id: &id_type::CustomerId,
+    payment_method_id: id_type::GlobalPaymentMethodId,
+    locker_id: Option<String>,
+    merchant_id: &id_type::MerchantId,
+    pm_metadata: Option<common_utils::pii::SecretSerdeValue>,
+    customer_acceptance: Option<common_utils::pii::SecretSerdeValue>,
+    payment_method_data: crypto::OptionalEncryptableValue,
+    key_store: &domain::MerchantKeyStore,
+    connector_mandate_details: Option<common_utils::pii::SecretSerdeValue>,
+    status: Option<enums::PaymentMethodStatus>,
+    network_transaction_id: Option<String>,
+    storage_scheme: MerchantStorageScheme,
+    payment_method_billing_address: crypto::OptionalEncryptableValue,
+    card_scheme: Option<String>,
 ) -> errors::CustomResult<domain::PaymentMethod, errors::ApiErrorResponse> {
-    todo!()
+    let db = &*state.store;
+    let client_secret = pm_types::PaymentMethodClientSecret::generate(&payment_method_id);
+    let current_time = common_utils::date_time::now();
+
+    let response = db
+        .insert_payment_method(
+            &state.into(),
+            key_store,
+            domain::PaymentMethod {
+                customer_id: customer_id.to_owned(),
+                merchant_id: merchant_id.to_owned(),
+                id: payment_method_id,
+                locker_id,
+                payment_method: Some(req.payment_method),
+                payment_method_type: Some(req.payment_method_type),
+                metadata: pm_metadata,
+                payment_method_data,
+                connector_mandate_details,
+                customer_acceptance,
+                client_secret: Some(client_secret),
+                status: status.unwrap_or(enums::PaymentMethodStatus::Active),
+                network_transaction_id: network_transaction_id.to_owned(),
+                created_at: current_time,
+                last_modified: current_time,
+                last_used_at: current_time,
+                payment_method_billing_address,
+                updated_by: None,
+                version: domain::consts::API_VERSION,
+                locker_fingerprint_id: None,
+            },
+            storage_scheme,
+        )
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed to add payment method in db")?;
+
+    Ok(response)
+}
+
+#[cfg(all(feature = "v2", feature = "payment_methods_v2"))]
+#[instrument(skip_all)]
+#[allow(clippy::too_many_arguments)]
+pub async fn create_payment_method_for_intent(
```

Comment:
Payment method related function should not be in cards.rs, instead it should be in payment_method.rs file